OpenDreamRuntime/Objects/DreamObject.cs

```diff
@@ -368,22 +368,61 @@ public class DreamObject {
 
         #region Operators
         // +
-        public virtual DreamValue OperatorAdd(DreamValue b) {
+        public virtual ProcStatus OperatorAdd(DreamValue b, DMProcState state, out DreamValue? result) {
```

Comment:
The `result` argument in this and all the other operators should be made not nullable since that can have a performance impact on struct values. `result = null` could be replaced with `default` or `DreamValue.Null`